views/user-form-edit-profile.twig

```diff
@@ -0,0 +1,195 @@
+                {% set helpText %}{% trans "Please repeat the new Password." %}{% endset %}
+                {{ forms.password("retypeNewPassword", title, "", helptext) }}
+
+                {% set title %}{% trans "Email" %}{% endset %}
+                {% set helpText %}{% trans "The Email Address for this user." %}{% endset %}
+                {{ forms.email("email", title, currentUser.email, helpText) }}
+
+                {% set title %}{% trans "Two Factor Authentication" %}{% endset %}
+                {% set helpText %}{% trans "With this option enabled, user will be required to provide 2FA authentication code to log in to the CMS" %}{% endset %}
+                {% set values = [{id: 0, value: "Off"}, {id: 1, value: "Email"}, {id: 2, value: "Google Authenticator"}] %}
+                {{ forms.dropdown("twoFactorTypeId", "single", title, currentUser.twoFactorTypeId, values, "id", "value", helpText) }}
+
+                {{ forms.hidden("twoFactorRecoveryCodes", currentUser.twoFactorRecoveryCodes) }}
+                <div id="qRCode">
+                    <p>{% trans "Please scan the following image with your app:" %}</p>
+                    <p id="qr"><img id="qrImage" src=""></p>
+                    <div>
+                        {% set title %}{% trans "Access Code" %}{% endset %}
+                        {{ forms.input("code", title) }}
+                    </div>
+                </div>
+
+                <div id="recoveryButtons">
+                    <p>{% trans "Please use the buttons below to generate or show your two factor recovery codes." %}</p>
+
+                    <ul class="nav nav-pills pull-left" style="margin-left: 5px;">
+                        <li class="btn btn-success btn-xs recBtn"><a class="btns generateCodes" id="generateCodesBtn" style="padding: 7px 7px;" title="{% trans "Generate Recovery codes, this action will invalidate all existing recovery codes." %}" > <i class="fa fa-plus-circle" aria-hidden="true"></i> {% trans "Generate" %}</a></li>
+                    </ul>
+
+                    <ul class="nav nav-pills pull-left" style="margin-left: 5px;">
+                        <li class="btn btn-info btn-xs recBtn"><a class="btns showCodes" id="showCodesBtn" style="padding: 7px 7px;" title="{% trans "Show existing recovery codes" %}" > <i class="fa fa-plus-circle" aria-hidden="true"></i> {% trans "Show" %}</a></li>
+                    </ul>
+                </div>
+
+                <div id="recoveryCodes" style="float: left; width: 100%;">
+                    <p>{% trans "Here are your recovery codes, please make sure to store them in a safe place, like password manager." %}</p>
+                    <p>{% trans "Recovery codes will become active only after this form is successfully saved" %}</p>
+                    <p id="codesList" class=""></p>
+                </div>
+            </form>
+        </div>
+    </div>
+
+    <script type="text/javascript">
+        $(document).ready(function() {
```

Comment:
I am quite surprised this works because this whole form is returned via AJAX. This should probably be a function in `authed.twig` which is then called on the form load callback?